Doc/library/sqlite3.rst

```diff
@@ -1279,45 +1284,21 @@ Row objects
    .. versionchanged:: 3.5
       Added support of slicing.
 
-Let's assume we initialize a table as in the example given above::
-
-   con = sqlite3.connect(":memory:")
-   cur = con.cursor()
-   cur.execute('''create table stocks
-   (date text, trans text, symbol text,
-    qty real, price real)''')
-   cur.execute("""insert into stocks
-               values ('2006-01-05','BUY','RHAT',100,35.14)""")
-   con.commit()
-   cur.close()
-
-Now we plug :class:`Row` in::
-
-   >>> con.row_factory = sqlite3.Row
-   >>> cur = con.cursor()
-   >>> cur.execute('select * from stocks')
-   <sqlite3.Cursor object at 0x7f4e7dd8fa80>
-   >>> r = cur.fetchone()
-   >>> type(r)
-   <class 'sqlite3.Row'>
-   >>> tuple(r)
-   ('2006-01-05', 'BUY', 'RHAT', 100.0, 35.14)
-   >>> len(r)
-   5
-   >>> r[2]
-   'RHAT'
-   >>> r.keys()
-   ['date', 'trans', 'symbol', 'qty', 'price']
-   >>> r['qty']
-   100.0
-   >>> for member in r:
-   ...     print(member)
-   ...
-   2006-01-05
-   BUY
-   RHAT
-   100.0
-   35.14
+   Example::
+
+      >>> con.row_factory = sqlite3.Row  # con is an sqlite3.Connection object.
+      >>> res = con.execute("select 'Tellus' as name, 6378 as radius")
```

Comment:
Yes, it uses a SELECT query with literals. The connection is needed to execute the query.

We can use another example; I just don't like the current example.

Reply:
We can create a test database as well, but that will expand the example with, IMO, unneeded information. We're interested in the row factory only; using a SELECT query that operates with literals quickly gets us a row of data without the need for a database table.

Reply:
> The connection is needed to execute the query.

Is this just a connection to `:memory:`?  Since I'm not querying an existing table, I assume that I don't need to create it and populate it -- how many lines would it take to create a valid `con`?

The goal for me would be having a reproducible example that I can play around with in the interactive interpreter.

Are there other factories?  Maybe the example could be moved back in an howto that sets up a single db and compares different factories if you think an howto specific to row factories is overkill?

Reply:
> Is this just a connection to :memory:? 

With the type of query used here, that does not matter. It could be any kind of connection, with or without tables. Since we're SELECTing literals, it does not matter.

A minimal connection is:

```python3
import sqlite3
cx = sqlite3.connect(":memory:")
```

Reply:
> Are there other factories? Maybe the example could be moved back in an howto that sets up a single db and compares different factories if you think an howto specific to row factories is overkill?

Not at the moment, but there has been talk about a namedtuple-factory.

I think a general "how to use/write row factories" could be useful. There is a current example in the `row_factory` reference; we could build on that. (For another PR/issue, of course.)

Reply:
> A minimal connection is:

Since the `import sqlite3` can be omitted, this means that a single line (`con = sqlite3.connect(":memory:")`) is enough to turn that into a working example, right?
If that's the case, having a working example outweighs the extra line IMHO.

The `SELECT` with literals is still a bit "weird" if someone is not familiar with it, but the full example will help showing that it's performed on an empty DB and is not assuming the existence of an existing table.  If you want to be more explicit, perhaps a footnote could be added.

Reply:
Yes, the SELECT trick may feel weird if you are unfamiliar with it. It may make sense to add a comment or footnote.

Reply:
A comment might probably be most appropriate, IMO, since it explains a potentially confusing but specific implementation detail of the example

Reply:
Hm, the SELECT-with-literals trick is very convenient for a lot of examples. Perhaps it would be better to add a "tips and tricks" howto, where we show it, and just reference that howto from the examples that use it?[^1] Obviously, it would not make sense to explain it every time we use it, hence my slight reluctance to add a note about it in this particular example.

[^1]: it could be a footnote too

Reply:
I'm not sure how to best resolve this _right now_. The old example also used SELECT-with-literals, so IMO, we're not introducing more confusion than there possibly was before. We can always improve matters with future PRs. I'll create a ticket for this on the tracker, though, so we don't forget about it.

Reply:
I created #96165